packages/mui-material/src/InputLabel/InputLabel.d.ts

```diff
@@ -74,4 +80,11 @@ export interface InputLabelProps extends StandardProps<FormLabelProps> {
  * - [InputLabel API](https://mui.com/material-ui/api/input-label/)
  * - inherits [FormLabel API](https://mui.com/material-ui/api/form-label/)
  */
-export default function InputLabel(props: InputLabelProps): JSX.Element;
+declare const InputLabel: OverridableComponent<InputLabelTypeMap>;
+
+export type InputLabelProps<
+  D extends React.ElementType = InputLabelTypeMap['defaultComponent'],
+  P = {},
+> = OverrideProps<InputLabelTypeMap<P, D>, D>;
```

Comment:
I think we won't need `OverrideProps` for `CommonProps` because `StandardProps` type already contains `className` and `style`.

Reply:
you mean like this?

```
export type InputLabelProps<
  D extends React.ElementType = InputLabelTypeMap['defaultComponent'],
  P = {},
> = InputLabelTypeMap<P, D>
```

Reply:
I thought so, but let's keep it in case a component is provided in the first generic parameter of `InputLabelProps` type. What we can do is remove the extension of `StandardProps` type from above.

Reply:
I'm not entirely sure about this, if StandardProps is removed then from where will InputLabel infer other types like (onClick,onChange...)

Reply:
this test would fail if StandardProps is removed

https://github.com/mui/material-ui/pull/35852/files#diff-f5beb3d0a6cd307137bc2c29e830648a962b649fb44ea6b0221f9d0360f8cadaR13